{"metadata": {"language_info": {"name": "python", "version": "3.9.23", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}, "kernelspec": {"name": "conda_py39", "display_name": "conda_py39", "language": "python"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "# ToDo List\n- Check WGS bgen files is phased?\n- Filter chr 22 bgen file for MAF > 1%\n    - From previous processed data, look out for a list of all snps with MAF > 1% on chromosome 22\n\n# TechnicalToDo List\n- Add cond.py39 env as a jupyter kernel\n- Install bgenix\n- rm all files from current env and save it as a new snapshot for future, add bgenix", "metadata": {}}, {"cell_type": "markdown", "source": "# Filtering bgen file when bgen.bgi file is availabe using plink2 is wrong because it is too slow\n# bgenix should be used with the bgen.bgi file for a real quick filteration", "metadata": {}}, {"cell_type": "code", "source": "# run in the terminal separately\n#conda init\n#exec #SHELL \n#conda activate py39\n#python -m ipykernel install --user --name=conda_py39 --display-name \"conda_py39\"", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "# kernel: conda_py39, where snipar=0.0.22 is installed\n# kernel added to jupyter kernels and selected upon creation of the current notebook", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": []}, {"cell_type": "code", "source": "# install bgenix\n# run in a separate terminal\n#conda activate py39 \n#conda install -c conda-forge bgenix", "metadata": {"trusted": true, "tags": []}, "execution_count": 15, "outputs": []}, {"cell_type": "code", "source": "!dx download \"project-J1VzYJQJQp6z67g2f0Vvb2kK:/Bulk/DRAGEN WGS/DRAGEN population level WGS variants, BGEN format [500k release]/ukb24309_c22_b0_v1.bgen\"", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 246,913,601,766 of 246,913,601,766 bytes (100%) /opt/notebooks/ukb24309_c22_b0_v1.bgenn\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "!dx download \"project-J1VzYJQJQp6z67g2f0Vvb2kK:/Bulk/DRAGEN WGS/DRAGEN population level WGS variants, BGEN format [500k release]/ukb24309_c22_b0_v1.bgen.bgi\"", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 1,310,730,240 of 1,310,730,240 bytes (100%) /opt/notebooks/ukb24309_c22_b0_v1.bgen.bgii\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# installing plink2 to filter bgen file\n# bash\n\n#wget https://s3.amazonaws.com/plink2-assets/alpha6/plink2_linux_x86_64_20250707.zip\n#unzip plink2_linux_x86_64_20250707.zip\n#mv plink2 /usr/local/bin/\n#plink2 --help", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": []}, {"cell_type": "code", "source": "#download c22.txt that contains ~2K random SNPs on chr22 with MAF > 1%\n!dx download project-J1VzYJQJQp6z67g2f0Vvb2kK:/user_data__Mahdi/i70_g6_CSF/b_inp/c22.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 136,295 of 136,295 bytes (100%) /opt/notebooks/c22.txtt\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# download sample for the bgen file\n!dx download \"project-J1VzYJQJQp6z67g2f0Vvb2kK:/Bulk/DRAGEN WGS/DRAGEN population level WGS variants, BGEN format [500k release]/ukb24309_c22_b0_v1.sample\"", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 13,674,451 of 13,674,451 bytes (100%) /opt/notebooks/ukb24309_c22_b0_v1.samplee\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# download agesex.txt to only keep those IDs present in it from the bgen file for the purpose of imputataion and FGWAS and also ibd inference\n!dx download project-J1VzYJQJQp6z67g2f0Vvb2kK:/user_data__Mahdi/i70_g6_CSF/c_med/agesex.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 1,315,222 of 1,315,222 bytes (100%) /opt/notebooks/agesex.txtt\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# check what type of IDs the BGEN file uses\n# !plink2 --bgen ukb24309_c22_b0_v1.bgen ref-first --sample \"ukb24309_c22_b0_v1.sample\" --write-snplist --out variant_ids\n\n# it is too slow", "metadata": {"trusted": true, "tags": []}, "execution_count": 14, "outputs": [{"name": "stdout", "text": "PLINK v2.0.0-a.6.20LM 64-bit Intel (7 Jul 2025)    cog-genomics.org/plink/2.0/\n(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to variant_ids.log.\nOptions in effect:\n  --bgen ukb24309_c22_b0_v1.bgen ref-first\n  --out variant_ids\n  --sample ukb24309_c22_b0_v1.sample\n  --write-snplist\n\nStart time: Mon Jul 21 17:29:02 2025\n70303 MiB RAM detected, ~67832 available; reserving 35151 MiB for main\nworkspace.\nUsing up to 36 threads (change this with --threads).\n--bgen: 19511731 variants declared in header, format v1.3.\n490541 samples imported from .sample file to variant_ids-temporary.psam .\n--bgen: 9k variants scanned.^C\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "!plink2 --bgen \"ukb24309_c22_b0_v1.bgen\" ref-first --export bgen-1.2 --sample \"ukb24309_c22_b0_v1.sample\" --keep \"agesex.txt\" --extract \"c22.txt\" --out \"wgs_chr22_filtered\"", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [{"name": "stdout", "text": "PLINK v2.0.0-a.6.20LM 64-bit Intel (7 Jul 2025)    cog-genomics.org/plink/2.0/\n(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\nLogging to wgs_chr22_filtered.log.\nOptions in effect:\n  --bgen ukb24309_c22_b0_v1.bgen ref-first\n  --export bgen-1.2\n  --extract c22.txt\n  --keep agesex.txt\n  --out wgs_chr22_filtered\n  --sample ukb24309_c22_b0_v1.sample\n\nStart time: Sun Jul 20 23:41:06 2025\n70303 MiB RAM detected, ~68075 available; reserving 35151 MiB for main\nworkspace.\nUsing up to 36 threads (change this with --threads).\n--bgen: 19511731 variants declared in header, format v1.3.\n490541 samples imported from .sample file to wgs_chr22_filtered-temporary.psam\n.\n--bgen: 1258k variants scanned.", "output_type": "stream"}]}]}