{"metadata": {"language_info": {"name": "bash", "codemirror_mode": "shell", "mimetype": "text/x-sh", "file_extension": ".sh"}, "kernelspec": {"name": "bash", "display_name": "Bash", "language": "bash"}}, "nbformat_minor": 4, "nbformat": 4, "cells": [{"cell_type": "code", "source": "#kernel: bash", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "# install bgenix if not installed\n# run in a separate terminal\n#conda activate py39 \n#conda install -c conda-forge bgenix", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": []}, {"cell_type": "code", "source": "conda init", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [{"name": "stdout", "text": "no change     /opt/conda/condabin/conda\nno change     /opt/conda/bin/conda\nno change     /opt/conda/bin/conda-env\nno change     /opt/conda/bin/activate\nno change     /opt/conda/bin/deactivate\nno change     /opt/conda/etc/profile.d/conda.sh\nno change     /opt/conda/etc/fish/conf.d/conda.fish\nno change     /opt/conda/shell/condabin/Conda.psm1\nno change     /opt/conda/shell/condabin/conda-hook.ps1\nno change     /opt/conda/lib/python3.12/site-packages/xontrib/conda.xsh\nno change     /opt/conda/etc/profile.d/conda.csh\nno change     /home/dnanexus/.bashrc\nNo action taken.\n\nNote: you may need to restart the kernel to use updated packages.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "#restart kernel using GUI", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": []}, {"cell_type": "code", "source": "conda activate py39", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "bgenix -help", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [{"name": "stdout", "text": "\nWelcome to bgenix\n(version: 1.1.7, revision )\n\n(C) 2009-2017 University of Oxford\n\nUsage: bgenix <options>\n\nOPTIONS:\nInput / output file options:\n                  -g <a>: Path of bgen file to operate on.  (An optional form where \"-g\" is omitted and the filename is \n                          specified as the first argument, i.e. bgenix <filename>, can also be used).\n                  -i <a>: Path of index file to use. If not specified, bgenix will look for an index file of the form '<f-\n                          ilename>.bgen.bgi'  where '<filename>.bgen' is the bgen file name specified by the -g option.\n              -table <a>: Specify the table (or view) that bgenix should read the file index from. This only affects rea-\n                          ding the index file.  The named table or view should have the same schema as the Variant table\n                          written by bgenix on index creation.  Defaults to \"Variant\".\n\nIndexing options:\n                -clobber: Specify that bgenix should overwrite existing index file if it exists.\n                  -index: Specify that bgenix should build an index for the BGEN file specified by the -g option.\n             -with-rowid: Create an index file that does not use the 'WITHOUT ROWID' feature. These are suitable for use\n                          with sqlite versions < 3.8.2, but may be less efficient.\n\nVariant selection options:\n  -excl-range <a> <b>...: Exclude variants in the specified genomic interval from the output. See the description of -in-\n                          cl-range for details.If this is specified multiple times, variants in any of the specified ran-\n                          ges will be excluded.\n  -excl-rsids <a> <b>...: Exclude variants with the specified rsid(s) from the output. See the description of -incl-rang-\n                          e for details.If this is specified multiple times, variants with any of the specified ids will\n                          be excluded.\n  -incl-range <a> <b>...: Include variants in the specified genomic interval in the output. (If the argument is the name\n                          of a valid readable file, the file will be opened and whitespace-separated rsids read from it \n                          instead.) Each interval must be of the form <chr>:<pos1>-<pos2> where <chr> is a chromosome id-\n                          entifier  and pos1 and pos2 are positions with pos2 >= pos1.  One of pos1 and pos2 can also be\n                          omitted, in which case the range extends to the start or end of the chromosome as appropriate.\n                          Position ranges are treated as closed (i.e. <pos1> and <pos2> are included in the range).If th-\n                          is is specified multiple times, variants in any of the specified ranges will be included.\n  -incl-rsids <a> <b>...: Include variants with the specified rsid(s) in the output. If the argument is the name of a va-\n                          lid readable file, the file will be opened and whitespace-separated rsids read from it instead.I-\n                          f this is specified multiple times, variants with any of the specified ids will be included.\n\nOutput options:\n  -compression-level <a>: Zlib compression level to use when transcoding to BGEN v1.1 format.  Defaults to \"9\".\n                   -list: Suppress BGEN output; instead output a list of variants.\n                    -v11: Transcode to BGEN v1.1 format.  (Currently, this is only supported if the input is in BGEN v1.2\n                          format with 8 bits per probability, all samples are diploid, and all variants biallelic).\n                    -vcf: Transcode to VCF format.  VCFs will have GP field (or 'HP' field for phased data), and a GT fi-\n                          eld inferred from the probabilities by threshholding.\n\n\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# list variants in the chr 22 bgen file\nbgenix -g ukb24309_c22_b0_v1.bgen -list > chr22_variants.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [{"name": "stdout", "text": "\nWelcome to bgenix\n(version: 1.1.7, revision )\n\n(C) 2009-2017 University of Oxford\n\nBuilding query                                              :  (19511731/?,15.9s,1229197.7/s)\n\nThank you for using bgenix.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# check on the file \n# see if it has a header?\nhead chr22_variants.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [{"name": "stdout", "text": "# bgenix: started 2025-07-21 18:01:00\nalternate_ids\trsid\tchromosome\tposition\tnumber_of_alleles\tfirst_allele\talternative_alleles\n.\tchr22:10510001:G:T\t22\t10510001\t2\tT\tG\n.\tchr22:10510002:A:G\t22\t10510002\t2\tG\tA\n.\tchr22:10510003:A:C\t22\t10510003\t2\tC\tA\n.\tchr22:10510003:A:G\t22\t10510003\t2\tG\tA\n.\tchr22:10510004:T:G\t22\t10510004\t2\tG\tT\n.\tchr22:10510005:T:A\t22\t10510005\t2\tA\tT\n.\tchr22:10510006:C:A\t22\t10510006\t2\tA\tC\n.\tchr22:10510006:C:T\t22\t10510006\t2\tT\tC\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# randomly select 2K rows of the output file, selecting 2K variants\nshuf -n 2000 chr22_variants.txt > chr22_variants_2k.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": []}, {"cell_type": "code", "source": "head chr22_variants_2k.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"name": "stdout", "text": ".\tchr22:50697500:G:T\t22\t50697500\t2\tT\tG\n.\tchr22:23130524:G:GAAAGAAAGAAAGA\t22\t23130524\t2\tGAAAGAAAGAAAGA\tG\n.\tchr22:25050084:A:G\t22\t25050084\t2\tG\tA\n.\tchr22:38802304:C:G\t22\t38802304\t2\tG\tC\n.\tchr22:29770609:TTAC:T\t22\t29770609\t2\tT\tTTAC\n.\tchr22:40144638:G:C\t22\t40144638\t2\tC\tG\n.\tchr22:23170831:AT:A\t22\t23170831\t2\tA\tAT\n.\tchr22:33909821:A:C\t22\t33909821\t2\tC\tA\n.\tchr22:34144766:C:A\t22\t34144766\t2\tA\tC\n.\tchr22:11946851:CA:C\t22\t11946851\t2\tC\tCA\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "# filter bgen file for this 2K variants\nbgenix -g ukb24309_c22_b0_v1.bgen -incl-rsids chr22_variants_2k.txt > wgs_chr22_filtered_2k.bgen", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "\nWelcome to bgenix\n(version: 1.1.7, revision )\n\n(C) 2009-2017 University of Oxford\n\nBuilding query                                              :  (2000/?,7.5s,265.8/s)\nProcessing 2000 variants                                    : [******************************] (2000/2000,5.3s,376.1/s)\nbgenix: wrote data for 2000 variants to stdout.\n\nThank you for using bgenix.\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "ls -lh wgs_chr22_filtered_2k.bgen", "metadata": {"trusted": true, "tags": []}, "execution_count": 10, "outputs": [{"name": "stdout", "text": "-rw-r--r-- 1 root root 34M Jul 21 18:31 wgs_chr22_filtered_2k.bgen\n", "output_type": "stream"}]}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}